### PR TITLE
[Feature] Add option to control the "cleared state" in Rules

### PR DESCRIPTION
**Pull request by shall0pass**
This adds the ability to control the "cleared" flag from the rules dialog by setting the value to true or false.

image

### CODE REVIEW

**Comment by MatissJanis** [comment]
I am able to create a rule that sets `cleared` to "hello". That does not seem right.

Screenshot 2023-01-18 at 22 09 55

**Comment by j-f1** [comment]
Maybe this should grab the checkbox from the transaction row and display itself as “set <kbd>cleared ⏷</kbd> to ☑️”?

**Comment by shall0pass** [comment]
I've added a checkbox instead of a text box to toggle the state.  Should have any data validation issues with that :)

**Comment by shall0pass** [comment]
I've setup a rule that "one of" "venmo, cash, or ca$happ" will result in a true cleared state.  When I enter a transfer between two of the three accounts I've specified, only 1 of the transactions receives the rule.  It appears the rules aren't applied to both transaction entries of a transfer between accounts.

File: packages/desktop-client/src/components/util/GenericInput.js [hunk 1]
```diff
@@ -149,6 +150,22 @@ export default function GenericInput({
...
+            <Checkbox
+              checked={value}
+              value={value}
+              onChange={e => onChange(!value)}
+            />
+          );
+          break;
+
+        default:
+      }
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/util/GenericInput.js:166]: **❓ question:** ‏why would we need a `switch` case here? IMO we can return the checkbox for all fields. And if sometime in the far future we need to have a different functionality for some other field - we can add it then. But until that time comes we can keep the code simple.

```suggestion
          content = (
            <Checkbox
              checked={value}
              value={value}
              onChange={e => onChange(!value)}
            />
          );
      }
```

Comment (shall0pass) [reply packages/desktop-client/src/components/util/GenericInput.js]: That's a great point.  I'll make the change.

**Comment by shall0pass** [comment]
I've learned a little more about the "cleared" state, at least as it pertains to transfers within Actual.  The rules are applied upon entering a transaction, therefore any of the rules applied to the first transaction will be applied to the newly created transaction for the transfer.  However, because the rules are already applied to the first transaction, they are never directly applied to the 2nd automatically created transaction.  Because of this logic, I think it would be wiser to tackle this issue in a different PR.

I proved this logic to myself by using this test case:
1.  Create a rule that adds a note to a transaction if the Account name is "Account 1".
2. Add a transfer from "Account 1" to "Account 2".  Results:  "Account 1" and "Account 2" both have the note.
3. Add a transfer from "Account 2" to "Account 1".  Results:  The note is not applied to either transaction.

I'm not making a case that this is wrong or right.  I can see a strong argument for the way it's currently implemented.  In the case of wanting to control the cleared status of an "Instant cleared" type account such as Venmo or Cash, it's a bit unfortunate.  

TLDR:  This doesn't work as expected for transfers between two accounts that have a "cleared rule" applied.

**Comment by MatissJanis** [comment]
@shall0pass your PR is not changing this functionality, right? So I'd say it's an existing limitation, but it should not block introducing the `cleared` field for rules.

**Review by MatissJanis — APPROVED** [review]
Code looks good 👍

**Comment by shall0pass** [comment]
> @shall0pass your PR is not changing this functionality, right? So I'd say it's an existing limitation, but it should not block introducing the `cleared` field for rules.

No. This code doesn't change that functionality.  It's just my observations when testing the different scenarios.

**Comment by MatissJanis** [comment]
Got it! if you're interested to pursue this further, then I'd encourage to open a bug report issue. Otherwise: are you happy if I merge this PR?

**Comment by shall0pass** [comment]
> Got it! if you're interested to pursue this further, then I'd encourage to open a bug report issue. Otherwise: are you happy if I merge this PR?

I'm happy with it.  Thanks!
